esmf_regrid/schemes.py

```diff
@@ -1,19 +1,315 @@
-"""TBD: public module docstring."""
+"""Provides an iris interface for regridding."""
+
+import copy
+
+import iris
+from iris.analysis._interpolation import get_xy_dim_coords
+import numpy as np
+
+from esmf_regrid.esmf_regridder import GridInfo, Regridder
 
 __all__ = [
     "ESMFAreaWeighted",
+    "ESMFAreaWeightedRegridder",
+    "regrid_rectilinear_to_rectilinear",
```

Comment:
Is this intended to be the final name / export of this library, or will the final regridder class abstract away the differences between passing in rectilinear or unstructured data?

Reply:
I think it's possible that ESMFAreaWeighted might be able to act in a way that is similar to the other two objects here. It is in a way the top level object which is why it was initially the only one included in `__all__`. With that said, I'm kind of going off my instructions for the unstructured case which involved presenting objects with interfaces like the two i have introduced here. I've included everything for the sake of completeness, though there is probably some redundancy here which I could imagine us tidying up in future.